### PR TITLE
Deploy model ranking from model config instance

### PR DESCRIPTION
**Pull request by thigm85**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: vespa/package.py [hunk 1]
```diff
@@ -889,6 +890,156 @@ def __init__(
...
+            Function(
+                name="token_type_ids",
+                expression="tensor<float>(d0[1],d1[{}])(\n"
+                "    if (d1 < question_length + 2,\n"
+                "        0,\n"
+                "    if (d1 < question_length + doc_length + 3,\n"
+                "        1,\n"
+                "        TOKEN_NONE\n"
+                "    )))".format(model_config.input_size),
+            ),
```
Comment (thigm85) [inline vespa/package.py:1011]: I will update `token_type_uds` to use `tokenTypeIds` as soon as https://github.com/vespa-engine/vespa/pull/16413 is available.

**Comment by thigm85** [comment]
update merge stop

File: vespa/package.py (hunk 1 above)
Comment (lesters) [reply vespa/package.py:1011]: Now, https://github.com/vespa-engine/vespa/pull/16432

File: vespa/package.py [hunk 2]
```diff
@@ -889,6 +890,156 @@ def __init__(
...
+        summary_features = [
+            "logit0",
+            "logit1",
+            "input_ids",
+            "attention_mask",
+            "token_type_ids",
+        ]
+        if "summary_features" in kwargs:
+            summary_features.extend(kwargs.pop("summary_features"))
+
```
Comment (lesters) [inline vespa/package.py:1033]: Not sure this is something we always want to add. Added logit0 and logit1 functions to summary features involves evaluating the model also during summary fetch time in addition to during ranking. Could this be an option?

**Review by lesters — APPROVED** [review]
LGTM. But consider comment regarding summary features.

File: vespa/package.py (hunk 1 above)
Comment (thigm85) [reply vespa/package.py:1031]: I will change that in a separate PR upon Vespa 7.355.9 release.

File: vespa/package.py (hunk 2 above)
Comment (thigm85) [reply vespa/package.py:1057]: We now have a `include_model_summary_features` argument in `add_model_ranking` that default to `False`.

**Pull request merged**